src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIUsesFurnace.java

```diff
@@ -64,7 +64,7 @@ protected AbstractEntityAIUsesFurnace(@NotNull final J job)
         super(job);
         super.registerTargets(
           new AITarget(IDLE, START_WORKING, STANDARD_DELAY),
-          new AITarget(START_WORKING, this::startWorking, TICKS_SECOND),
+          new AITarget(START_WORKING, this::startWorking, 100),
```

Comment:
Can we use the supplier pattern that Ray put together here, and use TICKS_SECOND if there is a current recipe set? The AI goes back to START_WORKING constantly during actual processing, so a 5 second delay will be fairly noticeable between actions.

Reply:
We could potentially have the supplier check both currentRecipe and the count of current furnaces burning. If we have no recipe, or burning furnaces, this could slow down quite a bit.

Reply:
What is the smeltercrafting doing that needs a higher update rate then every 5s?

Reply:
This is going to hit between picking up results for each furnace. So, with 5 furnaces you're looking at almost 30 seconds to retrieve results from all of them.

Reply:
That is just one example where we go back to START_WORKING in the AI. Most actions do.

Reply:
Not considering probably walking to the furnace too.

Reply:
Might be better to use a setDelay here then instead?

Reply:
Walking depends on the schematic, there are quite a few where there is little to no walking during pickup. SetDelay could work, but you added the provider support for just such an occasion to get away from the SetDelays.

Reply:
he barly does any actions besides waiting for the furnace and sometimes taking out stuff, his action are good if they take some time then he seems less idle all day. Its supposed to be a human-like those aren't instant robot-like doing all tasks at once

Reply:
Maybe a better question is, why is this call causing lag?

Reply:
I'm not disagreeing, @someaddons. There will be delays in what I'm proposing, more than there are today. She's going to be slower to pick up new jobs, and to react to furnaces finishing. All I'm hoping for is that once she starts to work, she doesn't take breaks every step.

Reply:
Heck, if we go to a provider, we can slow down this check by a LOT. As in, if we have a recipe and things are burning, then 20-30 seconds. If we just have a recipe, 1 second, if we have no recipe, 10-15 seconds.

Reply:
tbh I do not see delays between actions as breaks, to me its when the worker is "working" on sth that he takes some time. The reason why its relatively laggy are inventory searches

Reply:
I think the #1 solution should be not reducing the tick rate. But simply adding to the startWorking method, after the walkToBuilding a check if there is request in the queue and if not, directly exit and not do the 100 checks we do.

Reply:
sure you can redesign it if you want, but it is not necessary because a 5s wait is totally fine

Reply:
`The reason why its relatively laggy are inventory searches`

Ugh, why didn't you say so? that's an easy fix =P